src/main/java/org/pac4j/vertx/handler/impl/RequiresAuthenticationHandler.java

```diff
@@ -138,6 +145,33 @@ public void handle(RoutingContext routingContext) {
 
     }
 
+    @Override
+    protected void authorise(final User user, final RoutingContext context) {
+        if (! (user instanceof Pac4jUser)) {
+            throw new TechnicalException("Wrong user type in authorise");
+        }
+
+        Handler<AsyncResult<Boolean>> authHandler = res -> {
+            if (res.succeeded()) {
+                if (res.result()) {
+                    context.next();
+                } else {
+                    context.fail(403);
+                }
+            } else {
+                context.fail(res.cause());
```

Comment:
I'm not sure I understand the question but I'll try and answer on the assumption that I do - context.fail means we have an unexpected internal exception and will respond with a 500. It's the standard vert.x mode of handling an unexpected exception during request processing. And we'll only get a failed because of an unexpected runtime exception, so that would appear to be sensible to me.